orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -73,7 +73,7 @@ def forward(self, *inputs, **kwargs):
                                                       kwargs)
 
         # Reinitialize graph builder if the inputs or initializers requiring gradient have changed.
-        build_gradient_graph = build_gradient_graph or self._reinitialize_graph_builder(input_info)
+        build_gradient_graph = self._reinitialize_graph_builder(input_info) or build_gradient_graph
```

Comment:
Did this fix swap any subtle issue?

Reply:
What I realized here is that if ```build_gradient_graph``` was ```True```, then ```self._reinitialize_graph_builder(input_info)``` would not be called at all.
So, this swap is needed to ensure that ```self._reinitialize_graph_builder(input_info)``` is called even if ```build_gradient_graph``` is ```True```.